packages/amazon-pay-v2-integration/src/AmazonPayV2PaymentMethod.tsx

```diff
@@ -0,0 +1,98 @@
+import { PaymentInitializeOptions } from '@bigcommerce/checkout-sdk';
+import { some } from 'lodash';
+import React, { FunctionComponent, useCallback } from 'react';
+
+import { HostedWidgetPaymentComponent } from '@bigcommerce/checkout/hosted-widget-integration';
+import {
+    isInstrumentCardCodeRequiredSelector,
+    isInstrumentCardNumberRequiredSelector,
+} from '@bigcommerce/checkout/instrument-utils';
+import {
+    PaymentMethodProps,
+    PaymentMethodResolveId,
+    toResolvableComponent,
+} from '@bigcommerce/checkout/payment-integration-api';
+
+const AmazonPayV2PaymentMethod: FunctionComponent<PaymentMethodProps> = ({
+    checkoutService,
+    checkoutState,
+    paymentForm,
+    method,
+    method: {
+        initializationData: { paymentDescriptor, paymentToken },
+    },
+    ...rest
+}) => {
+    const initializeAmazonPayV2Payment = useCallback(
+        (options: PaymentInitializeOptions) =>
+            checkoutService.initializePayment({
+                ...options,
+                amazonpay: {
+                    editButtonId: 'editButtonId',
+                },
+            }),
+        [checkoutService],
+    );
+
+    const reload = useCallback(() => window.location.reload(), []);
+
+    const {
+        hidePaymentSubmitButton,
+        disableSubmit,
+        setFieldValue,
+        setSubmit,
+        setValidationSchema,
+    } = paymentForm;
+
+    const {
+        data: { getCheckout, isPaymentDataRequired },
+        statuses: { isLoadingInstruments },
+    } = checkoutState;
+
+    const checkout = getCheckout();
+    const customer = checkoutState.data.getCustomer();
+    const isGuestCustomer = customer?.isGuest;
+    const isInstrumentFeatureAvailable =
+        !isGuestCustomer && Boolean(method.config.isVaultingEnabled);
+
+    const instruments = checkoutState.data.getInstruments(method) || [];
+
+    return (
+        <HostedWidgetPaymentComponent
+            {...rest}
+            buttonId="editButtonId"
+            containerId="paymentWidget"
+            deinitializeCustomer={undefined}
```

Comment:
Why are u set the `undefined` value like this?

Reply:
This solution was copied with AmazonPayV2PaymentMethod from core, but after analyzing, looks like these props are not needed here to be pass as undefined. I removed them.